cmake/external/onnxruntime_external_deps.cmake

```diff
@@ -46,8 +46,8 @@ if (onnxruntime_BUILD_UNIT_TESTS)
   FetchContent_Declare(
     googletest
     URL ${DEP_URL_googletest}
+    FIND_PACKAGE_ARGS NAMES GTest
```

Comment:
FYI, from the build errors, it looks like the version of gtest that it is finding is too old. some of those errors look familiar from when we moved to MS GSL - those changes required a gtest upgrade. I believe gtest 1.12.1+ will work.